namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+
+
+# General Elements Definition
+
+General elements defined inside MSON namespaces but possibly reusable in another domain.
+
+## Identifier (Enum)
+
+Identifier of an element in the MSON namespace. The identifier MUST be either directly a `string` value or an `Id Element` element.
+
+### Members
+
+- (string)
+- (Id Element)
+
+## Id Element (Distinct Element)
+
+Element representing an identifier in the MSON namespace.
+
+### Properties
+
+- `element`: id (string, required, fixed)
+- `content` (string, required)
+
+## Distinct Element (Element)
+
+Abstract element representing an element that can be referenced or is a reference to another element.
+
+### Properties
+
+- `element` (Identifier, required) - identifier of this element's base element
+- `attributes`
+    - `id` (Identifier) - identifier of this element
+    - `ref` (Identifier) - reference to expanded element
+
+### Example
+
+```json
+{
+    "element": "number",
+    "attributes": {
+        "id": "my-number"
+    },
+    "content": 42
+}
+```
+
+which is identical to:
+
+```json
+{
+    "element": {
+        "element": "id",
+        "content": "number"
+    },
+    "attributes": {
+        "id": "my-number"
+    },
+    "content": 42
+}
+```
+
+and can be referred to as a _base element_:
+
+```json
+{
+    "element": "my-number"
+}
+```
+
+which _expands_ to:
+
+```json
+{
+    "element": "number",
+    "attributes": {
+        "ref": "my-number"
+    },
+    "content": 42
+}
+```
+
+## Select (Distinct Element)
+
+Element representing selection of options. Every item of content array represents one possible option.
+
+### Properties
+
+- `element`: select (string, required, fixed)
+- `content` (array[Option])
+
+## Option (Distinct Element)
+
+One choice in the selection.
+
+### Properties
+
+- `element`: option (string, required, fixed)
+- `content` (enum)
+    - (Distinct Element)
+    - (array[Distinct Element])
+
+#### Example
+
+```html
+<select>
+  <option>Volvo</option>
+  <option>Saab</option>
+</select>
+```
+
+```json
+{
+    "element": "select",
+    "content": [
+        {
+            "element": "option",
+            "content": [
+                {
+                    "element": "string",
+                    "content": "volvo"
+                }
+            ]
+        }
+        {
+            "element": "option",
+            "content": [
+                {
+                    "element": "string",
+                    "content": "Saab"
+                }
+            ]
+        }
+    ]
+}
+```
+
+# MSON DOM Elements Definition
+
+## MSON Element (Distinct Element)
+
+Base element for all MSON elements.
+
+The MSON Element adds attributes representing MSON _Type Definition_ and _Type Sections_. Note in MSON DOM _Nested Member Types_ _Type Section_ is the `content` of the element.
+
+### Properties
+
+- `attributes`
+    - `typeAttributes` (array) - _Type Definition_ attributes list, see _Type Attribute_  
+        - (enum[string])
```

Comment:
Since you weren't there when we made MSON spec: No, enums are treated as fixed by default.